jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/DataprocInfo.java

```diff
@@ -0,0 +1,169 @@
+                        params.transportMode(),
+                        params.httpPath());
+        if (params.otherSessionConfs() != null) {
+            hiveJdbcURL = String.format("%s;%s", hiveJdbcURL, params.otherSessionConfs());
+        }
+        if (params.hiveConfs() != null) {
+            hiveJdbcURL = String.format("%s?%s", hiveJdbcURL, params.hiveConfs());
+        }
+        if (params.hiveVars() != null) {
+            hiveJdbcURL = String.format("%s#%s", hiveJdbcURL, params.hiveVars());
+        }
+        return hiveJdbcURL;
+    }
+
+    /**
+     * Use Dataproc client libraries to get cluster.
+     *
+     * @return name of the cluster's master node
+     * @throws InvalidURLException
+     */
+    public String getHost() throws InvalidURLException {
+        String host;
+        if (params.clusterName() != null) {
+            host = getClusterByName();
+        } else {
+            // clusterPoolLabel can be null, but filter is not empty string since
+            // it at least has "status.state = RUNNING"
+            String filter = formatClusterFilterString();
+            host = findClusterInPool(filter);
+        }
+        return host + "-m";
+    }
+
+    /**
+     * Supports finding & verifying the cluster indicated by name.
+     *
+     * @return the cluster of the given name
+     * @throws InvalidURLException
+     */
+    public String getClusterByName() throws InvalidURLException {
+        try {
+            Cluster cluster =
+                    clusterControllerClient.getCluster(
+                            params.projectId(), params.region(), params.clusterName());
+            return cluster.getClusterName();
+        } catch (ApiException e) {
+            throw new InvalidURLException(
+                    String.format(
+                            "Unable to retrieve cluster information for %s/%s/%s.\n%s",
+                            params.projectId(), params.region(), params.clusterName(), e.getMessage()));
+        }
+    }
+
+    /**
+     * Formats the user passed in clusterPoolLabel value to the filter format that Dataproc API
+     * accpets: field = value [AND [field = value]] ...
+     *
+     * @return formatted filter string accepted by Dataproc Client API
+     */
+    public String formatClusterFilterString() throws InvalidURLException {
+        Map<String, List<String>> labels = new LinkedHashMap<>();
+        labels.put("status.state", new ArrayList<>(Arrays.asList("ACTIVE")));
+
+        if (params.clusterPoolLabel() != null) {
+            for (String label : params.clusterPoolLabel().split(":")) {
+                checkUrl(
+                        !label.isEmpty() && label.contains("=") && !label.startsWith("status"),
+                        "'%s' Invalid clusterPoolLabel.\n"
+                                + "Please do not provide cluster status as label.\n"
+                                + "Example format: clusterPoolLabel=field1=value1:field2=value2'",
```

Comment:
Sorry I didn't see this the first time, but you should add a test for what happens if the user passes in an invalid label like `foo` (with no `=` sign). You will need to add validation before parsing field/value.

You could simplify this conditional block by validation and parsing field/value first, and then after that having an if statement like:

```
if (field.equals("status")) {
  // throw exception
}
```